dist/index.js

```diff
@@ -55889,47 +55902,62 @@ var __webpack_exports__ = {}
         }
       })
       .filter((x) => !!x)
-  }
+  } // CONCATENATED MODULE: ./src/parsers/index.js
+
+  const parsers_parseDate = date_parseDate
+  const parsers_parseTime = time_parseTime
+  const parsers_parseDuration = parseDuration
+  const parsers_parseList = parseList // CONCATENATED MODULE: ./src/parse.js
 
   async function parseMD(body) {
     const tokens = await unified().use(remark_parse).use(remarkGfm).parse(body)
     if (!tokens) {
       return []
     }
 
-    const r = []
+    const r = {}
+    let counter = 0
     for (let idx = 0; idx < tokens.children.length; idx = idx + 2) {
       const current = tokens.children[idx]
       const hasNext = idx + 1 < tokens.children.length
 
       if (current.type === 'heading') {
-        // issue-form answers start with a h3 heading, ignore everything else
+        const key = slugify(current.children[0].value)
+
+        // issue-form answers start with a h3 heading, ignore everything else for now
         const obj = {
-          id: slugify(current.children[0].value),
-          title: current.children[0].value
+          title: current.children[0].value,
+          order: counter++
         }
+
         if (hasNext) {
           const next = tokens.children[idx + 1]
           if (next.type === 'list') {
-            obj.list = parseList(next).flat()
+            obj.list = parsers_parseList(next).flat()
           }
-          obj.text = await unified()
+          const text = await unified()
             .use(remarkGfm)
             .use(remark_stringify)
             .stringify(next)
-          const date = parse_parseDate(obj.text)
-          const time = parse_parseTime(obj.text)
+          obj.text = stripFinalNewline(text)
+
+          const date = parsers_parseDate(obj.text)
+          const time = parsers_parseTime(obj.text)
+
           if (date) {
             obj.date = date
           }
+
           if (time) {
             obj.time = time
           }
-          if (obj.id === 'duration') {
-            obj.duration = parseDuration(obj.text)
+
+          if (key === 'duration') {
+            obj.duration = parsers_parseDuration(obj.text)
           }
+
+          r[key] = obj
```

Comment:
So, I want to be very clear I'm advocating this should be an either-or between the 2 approaches: array of objects versus single object with multiple properties.

That said, do you think this is the right thing?  I know you said these changes would result in a `2.0.0` release as it would be breaking; just making sure.

Reply:
That's correct. `r[key]` sets the id for an object, not an array (looks a bit misleading). Check out the expected result in  the test: https://github.com/zentered/issue-forms-body-parser/blob/6-return-type/test/parse-issue.test.js#L11

Reply:
(terrible naming .... will change that at some point)